api/python/quilt3/util.py

```diff
@@ -280,9 +280,8 @@ def write_yaml(data, yaml_path, keep_backup=False):
     try:
         if path.exists():
             path.rename(backup_path)
-        if not path.parent.exists():
-            path.parent.mkdir(parents=True)
-        with path.open('w') as config_file:
+        path.parent.mkdir(parents=True, exist_ok=True)
+        with path.open('x') as config_file:
```

Comment:
This will raise an exception if the file exists. Ok, better than two processes writing to the same file, but... now look at the exception handler, and imagine what kind of race conditions happen there :)